conjur/logic/init_logic.py

```diff
@@ -42,8 +45,17 @@ def get_certificate(self, hostname:str, port):
         try:
             fingerprint, readable_certificate = self.ssl_service.get_certificate(hostname, port)
             logging.debug("Successfully fetched certificate")
+        except socket.gaierror as error:
+            raise ConnectionToConjurFailedException(f"Unable to resolve server DNS "
+                            f"from {hostname}:{port}. "
+                            f"Reason: {str(error)}") from error
+        except socket.timeout as error:
+            raise ConnectionToConjurFailedException(f"Unable to connect to server "
+                            f"from {hostname}:{port}. "
+                            f"Reason: {str(error)}") from error
         except Exception as error:
-            raise Exception(f"Unable to retrieve certificate from {hostname}:{port}. "
+            raise RetrieveCertificateException(f"Unable to retrieve certificate "
+                            f"from {hostname}:{port}. "
```

Comment:
can you actually run the flows and provide how the error messages look for each of these working flows? I want to make sure that actually running them will give us the output we want. For example:

gaierror error -> "blahblah"

Reply:
*Bad DNS
Failed to execute command. Reason: Unable to resolve server DNS from onefds.co.il:443. Reason: [Errno 8] nodename nor servname provided, or not known
Run the command again in debug mode for more information.

*Unable to connect
Failed to execute command. Reason: Unable to connect to server from ec2-3-1-1-111.compute-1.amazonaws.com:443. Reason: [Errno 60] Operation timed out
Run the command again in debug mode for more information.